dual_arm_teleop/src/robot_arm.cpp

```diff
@@ -0,0 +1,297 @@
+  finger_c.position = msg_controller.joystick.axes[2];
+
+  msg_out_gripper.scissor_command = scissor;
+  msg_out_gripper.finger_a_command = finger_a;
+  msg_out_gripper.finger_b_command = finger_b;
+  msg_out_gripper.finger_c_command = finger_c;
+
+  // Publish state messages
+  if (armWithinDelta(jvqToVector(msg_out_arm.joint_position))) {
+    pub_arm.publish(msg_out_arm);
+  }
+
+  pub_gripper.publish(msg_out_gripper);
+
+//  // Display rviz poses
+//  tf::Transform transform1;
+//  tf::poseEigenToTF(relative_pose, transform1);
+//  tf_broadcaster.sendTransform(tf::StampedTransform(transform1, ros::Time::now(), "victor_root", joint_model_group->getName() + "/relative_pose"));
+//
+//  tf::Transform transform2;
+//  tf::poseEigenToTF(translationAndRotationToAffine(controller_start_translation, controller_start_rotation), transform2);
+//  tf_broadcaster.sendTransform(tf::StampedTransform(transform2, ros::Time::now(), "victor_root", joint_model_group->getName() + "/reset_pose"));
+//
+//  tf::Transform transform3;
+//  tf::poseEigenToTF(poseMsgToEigen(msg_controller.posestamped.pose), transform3);
+//  tf_broadcaster.sendTransform(tf::StampedTransform(transform3, ros::Time::now(), "victor_root", joint_model_group->getName() + "/global_pose"));
+//
+//  tf::Transform transform4;
+//  tf::poseEigenToTF(last_valid_pose, transform4);
+//  tf_broadcaster.sendTransform(tf::StampedTransform(transform4, ros::Time::now(), "victor_root", joint_model_group->getName() + "/last_valid_pose"));
+//
+//  tf::Transform transform5;
+//  tf::poseEigenToTF(translationAndRotationToAffine(ee_start_translation, Eigen::Quaterniond::Identity()), transform5);
+//  tf_broadcaster.sendTransform(tf::StampedTransform(transform5, ros::Time::now(), "victor_root", joint_model_group->getName() + "/ee_start_translation"));
+}
+
+void RobotArm::updateMeasuredState(victor_hardware_interface::MotionStatus msg) {
+  joint_position_measured = jvqToVector(msg.measured_joint_position);
+}
+
+bool RobotArm::armWithinDelta(std::vector<double> joint_position_commanded)
+{
+  assert(joint_position_commanded.size() == joint_position_measured.size());
+
+  double distance = 0;
+
+  for (int i = 0; i < joint_position_commanded.size(); ++i) {
+    distance += pow(joint_position_commanded[i] - joint_position_measured[i], 2);
+  }
+
+  distance = sqrt(distance);
+
+  std::cout << "Joint space error for " << joint_model_group->getName() << ": " << distance << std::endl;
+
+  return distance < .7;
+}
+
+Eigen::Affine3d RobotArm::translationAndRotationToAffine(Eigen::Vector3d translation, Eigen::Quaterniond rotation)
+{
+  Eigen::Affine3d out = Eigen::Affine3d::Identity();
+  out.translate(translation);
+  out.rotate(rotation);
+  return out;
+}
+
+Eigen::Vector3d RobotArm::viveToVictorTranslation(Eigen::Vector3d vive)
+{
+  return Eigen::Vector3d(
+      -vive[2],
+      -vive[0],
+      vive[1]
+  );
+}
+
+Eigen::Quaterniond RobotArm::viveToVictorRotation(Eigen::Quaterniond vive)
```

Comment:
Yikes! This makes me very nervous.
At least put some comments in explaining why you negated and shuffled these.
I don't think this is a valid rotation quaternion though, since you shift the position of the `w`. 
I'd much rather see something like

```
Eigen::Quaterniond rotation(angle_axis_type)
return rotation*vive
```
https://eigen.tuxfamily.org/dox/classEigen_1_1Quaternion.html